apps/nsq_tail/nsq_tail.go

```diff
@@ -18,28 +18,38 @@ import (
 var (
 	showVersion = flag.Bool("version", false, "print version string")
 
-	topic         = flag.String("topic", "", "NSQ topic")
 	channel       = flag.String("channel", "", "NSQ channel")
 	maxInFlight   = flag.Int("max-in-flight", 200, "max number of messages to allow in flight")
 	totalMessages = flag.Int("n", 0, "total messages to show (will wait if starved)")
 
 	nsqdTCPAddrs     = app.StringArray{}
 	lookupdHTTPAddrs = app.StringArray{}
+	topics           = app.StringArray{}
 )
 
 func init() {
 	flag.Var(&nsqdTCPAddrs, "nsqd-tcp-address", "nsqd TCP address (may be given multiple times)")
 	flag.Var(&lookupdHTTPAddrs, "lookupd-http-address", "lookupd HTTP address (may be given multiple times)")
+	flag.Var(&topics, "topic", "NSQ topic (may be given multiple times)")
 }
 
 type TailHandler struct {
+	topicName string
 	totalMessages int
 	messagesShown int
 }
 
 func (th *TailHandler) HandleMessage(m *nsq.Message) error {
 	th.messagesShown++
-	_, err := os.Stdout.Write(m.Body)
+	_, err := os.Stdout.WriteString(th.topicName)
+	if err != nil {
+		log.Fatalf("ERROR: failed to write to os.Stdout - %s", err)
+	}
+	_, err = os.Stdout.WriteString(" | ")
+	if err != nil {
+		log.Fatalf("ERROR: failed to write to os.Stdout - %s", err)
+	}
+	_, err = os.Stdout.Write(m.Body)
```

Comment:
A side effect of this change is that the output can no longer be (easily) piped into subsequent processes, a common operation when using `nsq_tail`.

I suppose one option is to just multiplex the output without prefixing the topic name?